plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchRequestProcessor.java

```diff
@@ -273,13 +280,52 @@ public void onFailure(Exception e) {
                 }
             }
 
-            private void updateIncomeQueryObject(Object incomeQueryObject, Map<String, String> outputMapping, MLOutput mlOutput) {
+            /**
+             * Updates the income query object with values from the ML output based on the provided output mapping.
+             *
+             * This method iterates through the output mapping, retrieves corresponding values from the ML output,
+             * and updates the income query object accordingly. It also handles nested JSON structures and updates
+             * the request context with the new values.
+             *
+             * @param incomeQueryObject The object representing the income query to be updated.
+             * @param outputMapping A map containing the mapping between new query fields and model output field names.
+             * @param mlOutput The MLOutput object containing the results from the machine learning model.
+             * @param requestContext The context object for the current pipeline processing request.
+             *
+             * @throws IllegalArgumentException If a specified JSON path cannot be found in the query string.
+             *
+             * @implNote This method uses JsonPath for JSON manipulation and supports both regular and extended (ext) fields.
+             *           For extended fields, it creates nested structures if they don't exist.
+             *           The method also updates the request context with new field values for further processing.
+             *
+             * @see JsonPath
+             * @see PipelineProcessingContext
+             * @see MLOutput
+             */
+            private void updateIncomeQueryObject(
+                Object incomeQueryObject,
+                Map<String, String> outputMapping,
+                MLOutput mlOutput,
+                PipelineProcessingContext requestContext
+            ) {
                 for (Map.Entry<String, String> outputMapEntry : outputMapping.entrySet()) {
-                    String newQueryField = outputMapEntry.getKey();
-                    String modelOutputFieldName = outputMapEntry.getValue();
-                    Object modelOutputValue = getModelOutputValue(mlOutput, modelOutputFieldName, ignoreMissing, fullResponsePath);
-                    String jsonPathExpression = "$." + newQueryField;
-                    JsonPath.parse(incomeQueryObject).set(jsonPathExpression, modelOutputValue);
+                    String newQueryField = null;
```

Comment:
this declaration can be moved inside the try/catch block

Reply:
if moved inside the catch block then when throw exception, we cannot get the queryField information in the exception message